### ISSUE TITLE
Addition of `IRFFTN` Function to tensorflow 

### ISSUE THREAD

**Issue by rajveer43** [tensorflow/tensorflow#62313]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

latest

### Custom code

Yes

### Current behavior?

𝐓𝐞𝐧𝐬𝐨𝐫𝐅𝐥𝐨𝐰 lacks a built-in 𝐑𝐅𝐅𝐓𝐍 function, which is essential for performing real-to-complex Fast Fourier Transforms (FFT) along multiple axes. This means users face limitations when working with real-valued data in the frequency domain.

`irfftn` performs the inverse Fourier transform, which involves reversing the process applied by the forward FFT. It computes the spatial or time-domain representation of the data from its frequency-domain representation. My reuqest is to add this function to the tensorflow backend as it lacks the support for inverse rfftn function. 

below is the list of deep learning frameworks already have it this function natively supported.

[torch](https://pytorch.org/docs/stable/generated/torch.fft.irfftn.html)
[JAX](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.fft.irfftn.html)
[paddlepaddle](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/fft/irfftn_en.html)

**Comment by KartikeyBartwal** [comment]
Kindly share some more details so that I can get better clarity on the feature you are requesting

**Comment by rajveer43** [comment]
> Kindly share some more details so that I can get better clarity on the feature you are requesting

I added the necessary description here, kindly acknowledge that

**Comment by tilakrayal** [comment]
@rajveer43,
Could you please have a look at this official document where the **irfft** performs Inverse real-valued fast Fourier transform and let us know if you are looking for this feature.
https://www.tensorflow.org/api_docs/python/tf/signal/irfft
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/base_api/api_def_IRFFT.pbtxt#L22
https://github.com/tensorflow/tensorflow/commit/6f08eea0a53ec79717824677c5da915297f0b09c

Thank you!

**Comment by rajveer43** [comment]
I already used it, and I found adding `irfftn` adding to tensorflow backend,becuase it provides irfftn for multiple axis,

**Comment by sachinprasadhs** [comment]
As per the `pyTorch` implementation also I don't see axis implementation `torch.fft.irfft(input, n=None, dim=-1, norm=None, *, out=None)`.
Could you please clarify with toy example about what different you are expecting other than the existing TensorFlow implementation of `tf.signal.irfft`

**Comment by rajveer43** [comment]
sure I will do tgat
